MoreLinq/Return.cs

```
@@ -0,0 +1,106 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Mitch Bodmer. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System.Collections;
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+
+    partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Returns a single-element sequence containing the item provided.
+        /// </summary>
+        /// <typeparam name="T">The type of the item provided.</typeparam>
+        /// <param name="item">The item to wrap in a sequence.</param>
+        /// <returns>A sequence containing only the <paramref name="item"/>.</returns>
+
+        public static IEnumerable<T> Return<T>(T item) => new SingleElementList<T>(item);
+
+        internal sealed class SingleElementList<T> : IList<T>, IReadOnlyList<T>
+        {
+            private readonly T _item;
+
+            public SingleElementList(T item) => _item = item;
+
+            public IEnumerator<T> GetEnumerator() => new SingleElementEnumerator(_item);
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+            public void Add(T item) => throw new NotSupportedException("Single element list is immutable");
+
+            public void Clear() => throw new NotSupportedException("Single element list is immutable");
+
+            public bool Contains(T item) => _item.Equals(item);
```

This will throw if `_item` is `null`. Please add a test to prove so, let it fail and then fix the bug.